src/main/remote-sync/RemoteSyncManager.ts

```diff
@@ -0,0 +1,347 @@
+import Logger from 'electron-log';
+import * as helpers from './helpers';
+import {
+  RemoteSyncStatus,
+  RemoteSyncedFolder,
+  RemoteSyncedFile,
+  SyncConfig,
+  SYNC_OFFSET_MS,
+} from './helpers';
+import { reportError } from '../bug-report/service';
+import { obtainToken } from '../auth/service';
+import { DatabaseAdapter } from '../database/adapters/base';
+
+export class RemoteSyncManager {
+  private foldersSyncStatus: RemoteSyncStatus = 'IDLE';
+  private filesSyncStatus: RemoteSyncStatus = 'IDLE';
+  private status: RemoteSyncStatus = 'IDLE';
+  private onStatusChangeCallback: (newStatus: RemoteSyncStatus) => void =
+    () => {
+      return;
+    };
+  constructor(
+    private db: {
+      files: DatabaseAdapter<RemoteSyncedFile>;
+      folders: DatabaseAdapter<RemoteSyncedFolder>;
+    },
+    private config: {
+      fetchFilesLimitPerRequest: number;
+      fetchFoldersLimitPerRequest: number;
+      syncFiles: boolean;
+      syncFolders: boolean;
+    }
+  ) {}
+
+  onStatusChange(callback: (newStatus: RemoteSyncStatus) => void) {
+    this.onStatusChangeCallback = callback;
+  }
+  getSyncStatus(): RemoteSyncStatus {
+    return this.status;
+  }
+
+  isSynced() {
+    return this.status === 'SYNCED';
+  }
+
+  /**
+   * Triggers a remote sync so we can populate the localDB, this sync
+   * is global and starts pulling all the files the user has in remote.
+   *
+   * Throws an error if there's a sync in progress for this class instance
+   */
+  async startRemoteSync() {
```

Comment:
Will add an `smokeTest` here:
- Check the database is okay and exists

So, as the current sync, we ensure that nothing strange is going on when we try to sync something

Reply:
This is a promise, when you call .connect() on the database adapter, it can throw an error if something goes wrong, will fail to start this in that case

Reply:
I added an smokeTest function that checks for the RemoteSyncManager status, database issues will be handled by the database adapter in the connect() method. Anyways database operations are success or die, if something goes wrong during database insert, the sync fails